exotica/src/Problems/UnconstrainedTimeIndexedProblem.cpp

```diff
@@ -154,13 +170,25 @@ void UnconstrainedTimeIndexedProblem::Update(Eigen::VectorXdRefConst x, int t)
     }
 
     scene_->Update(x, static_cast<double>(t) * tau);
+    Phi[t].setZero(PhiN);
+    J[t].setZero();
     for (int i = 0; i < NumTasks; i++)
     {
         // Only update TaskMap if Rho is not 0
         if (Rho[t](i) != 0)
             Tasks[i]->update(x, Phi[t].data.segment(Tasks[i]->Start, Tasks[i]->Length), J[t].middleRows(Tasks[i]->StartJ, Tasks[i]->LengthJ));
     }
-    ydiff[t] = y[t] - Phi[t];
+    ydiff[t] = Phi[t] - y[t];
+}
+
+double UnconstrainedTimeIndexedProblem::getScalarCost(int t)
+{
+    return ydiff[t].transpose()*S[t]*ydiff[t];
```

Comment:
Make a check whether t is in the valid bounds and throw_pretty otherwise - see e.g. the getRho and setRho - I added it there because it's easy to accidentally be off by one and you don't want the Python kernel with all its state die because of that.